ras_backstage/controllers/survey_controller.py

```diff
@@ -83,3 +83,23 @@ def get_survey_ci_classifier(survey_id):
                  ci_selector=ci_selector['id'])
 
     return response.json()
+
+
+def update_survey_details(survey_ref, updated_survey_details):
+    logger.debug('Updating survey details', survey_ref=survey_ref)
+    url = f'{app.config["RM_SURVEY_SERVICE"]}surveys/ref/{survey_ref}'
+    payload = {
+        "ShortName": updated_survey_details['short_name'],
+        "LongName": updated_survey_details['long_name']
+    }
+
+    response = request_handler('PUT', url, auth=app.config['BASIC_AUTH'], json=payload)
+
+    if response.status_code == 404:
+        logger.error('Error retrieving survey details', survey_ref=survey_ref)
```

Comment:
Should this be logged at error level?

Reply:
We throw an ApiError right after so probably yes? If we get this response something must have gone wrong

Reply:
If that's the case we should have error level logs in a lot more places. But I don't think a 404 should result in an error log because it gives us a splunk alert for something that does not require an action. warning level maybe better?